hardware/OnkyoAVTCP.cpp

```diff
@@ -305,26 +356,157 @@ void OnkyoAVTCP::ReceiveSwitchMsg(const char *pData, int Len, bool muting, int I
 	gswitch.rssi = 12;
 	gswitch.seqnbr = 0;
 
-	if (result.empty()) {
-		m_mainworker.PushAndWaitRxMessage(this, (const unsigned char *)&gswitch, switch_types[ID].name, 255);
-		m_sql.safe_query("UPDATE DeviceStatus SET SwitchType=%d, CustomImage=%d WHERE(HardwareID == %d) AND (DeviceID == '%08x')", switch_types[ID].switchType, switch_types[ID].customImage, m_HwdID, ID);
-		if (switch_types[ID].options) {
-			result = m_sql.safe_query("SELECT ID FROM DeviceStatus WHERE (HardwareID==%d) AND (DeviceID=='%08x') AND (Type==%d) ", m_HwdID, ID, pTypeGeneralSwitch);
-			if (result.size() > 0) {
-				std::string Idx = result[0][0];
-				m_sql.SetDeviceOptions(atoi(Idx.c_str()), m_sql.BuildDeviceOptions(switch_types[ID].options, false));
+	sDecodeRXMessage(this, (const unsigned char *)&gswitch, switch_types[ID].name, 255);
```

Comment:
Please use the default SendSwitch function

Reply:
Ah yes, I think I can so that now that I've made the device creation saner. Will fix.

Reply:
Hm, maybe not... that sends a lighting2 message, and this is a generalswitch. I also do still have special magic to do, to look up the target level for the selector from LevelActions. I'm not quite sure I see how to make it use SendSwitch, or at least I don't see that it simplifies anything. Am I missing something?

Reply:
Does it matter is it is a lighting2 or general switch ? Or are you using a selector switch ?
There might be a SendSelectorSwitch option soon, till then please use you current method

Reply:
I have selector, on/off and level switches. Currently they are all `pTypeGeneralSwitch` devices and handled generically, based on the following table:

```
	{ "MVL", "AMT", "Master volume", STYPE_Dimmer, sSwitchGeneralSwitch, 8, NULL },
	{ "ZVL", "ZMT", "Zone 2 volume", STYPE_Dimmer, sSwitchGeneralSwitch, 8, NULL },
	{ "VL3", "MT3", "Zone 3 volume", STYPE_Dimmer, sSwitchGeneralSwitch, 8, NULL },
	{ "VL4", "MT4", "Zone 4 volume", STYPE_Dimmer, sSwitchGeneralSwitch, 8, NULL },
	{ "PWR", NULL, "Master power", STYPE_OnOff, sSwitchGeneralSwitch, 5, NULL },
	{ "ZPW", NULL, "Zone 2 power", STYPE_OnOff, sSwitchGeneralSwitch, 5, NULL },
	{ "PW3", NULL, "Zone 3 power", STYPE_OnOff, sSwitchGeneralSwitch, 5, NULL },
	{ "PW4", NULL, "Zone 4 power", STYPE_OnOff, sSwitchGeneralSwitch, 5, NULL },
	{ "SLI", NULL, "Master selector", STYPE_Selector, sSwitchTypeSelector, 5, NULL },
	{ "SLZ", NULL, "Zone 2 selector", STYPE_Selector, sSwitchTypeSelector, 5, NULL },
	{ "SL2", NULL, "Zone 3 selector", STYPE_Selector, sSwitchTypeSelector, 5, NULL },
	{ "SL3", NULL, "Zone 4 selector", STYPE_Selector, sSwitchTypeSelector, 5, NULL },
	{ "HDO", NULL, "HDMI Output", STYPE_Selector, sSwitchTypeSelector, 5, "SelectorStyle:0;LevelNames:Off|Main|Sub|Main+Sub;LevelOffHidden:true;LevelActions:00|01|02|03" },
```

Reply:
For now please keep using your direct calls. I think there will be a function to send/update selector switches shortly, maybe then we can replace these functions